### PR TITLE
MIG-10: Migrate product drafts and proposals

### PR DESCRIPTION
**Pull request by momoss**
**Description**

**Definition Of Done**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Ok
| Review and 2 GTM                  | Ok
| Micro Demo to the PO (Story only) | Ok

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Domain/MigrationStep/s160_ProductDraftMigration/ProductDraftImporter.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public function import(DestinationPim $pim, string $draftFilename)
+    {
+        $reader = ReaderFactory::create(Type::CSV);
+
+        $varDir = __DIR__ . '/../../../../var';
+        $reader->open($varDir .'/'. $draftFilename);
+
+        foreach ($reader->getSheetIterator() as $sheet) {
+            $hasHeader = true;
+            foreach ($sheet->getRowIterator() as $row) {
```
Comment (LaurentPetard) [inline src/Domain/MigrationStep/s160_ProductDraftMigration/ProductDraftImporter.php]: I think there's a mistake. $hasHeader is set to false at the first iteration, so why doing a `foreach` ?

File: src/Domain/MigrationStep/s160_ProductDraftMigration/ProductDraftImporter.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                if (!$hasHeader) {
+                    $sqlSelectIdQuery = 'SELECT id FROM `pim_catalog_product` WHERE identifier = "%s"';
+                    $id = $this->console->execute(new MySqlQueryCommand(sprintf($sqlSelectIdQuery, $row[0])), $pim)->getOutput()[0]['id'];
+
+                    $sqlInsertDraftQuery = "INSERT INTO `pimee_workflow_product_draft` (product_id, created_at, changes, status, author) VALUES (%s, '%s', '%s', %s, '%s')";
```
Comment (LaurentPetard) [inline src/Domain/MigrationStep/s160_ProductDraftMigration/ProductDraftImporter.php]: shouldn't the value of `changes` be escaped ?

Comment (momoss) [reply src/Domain/MigrationStep/s160_ProductDraftMigration/ProductDraftImporter.php]: They are!

File: src/Domain/MigrationStep/s160_ProductDraftMigration/config/Akeneo/Bundle/MigrationBundle/Command/ProductDraftMigrationCommand.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ */
+class ProductDraftMigrationCommand extends ContainerAwareCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('transporteo:migration:draft')
```
Comment (anaelChardan) [inline src/Domain/MigrationStep/s160_ProductDraftMigration/config/Akeneo/Bundle/MigrationBundle/Command/ProductDraftMigrationCommand.php]: Why transporteo if it is on the Pim?

**Pull request closed without merging**